src/main/java/org/ballerinax/datamapper/DataMapperStructureVisitor.java

```diff
@@ -0,0 +1,139 @@
+        int count = 0;
+        for (Iterator<BLangExpression> iterator = exprList.iterator(); iterator.hasNext(); ) {
+            BLangExpression expr = iterator.next();
+            if (expr.getKind() == NodeKind.LITERAL) {
+                expr.accept(this);
+            } else if (expr.getKind() == NodeKind.NAMED_ARGS_EXPR) {
+                expr.accept(this);
+            } else if (expr.getKind() == NodeKind.RECORD_LITERAL_EXPR) {
+                expr.accept(this);
+            } else if (expr.getKind() == NodeKind.LIST_CONSTRUCTOR_EXPR) {
+                expr.accept(this);
+            }
+            count += 1;
+        }
+    }
+
+    @Override
+    public void visit(BLangNamedArgsExpression varRefExpr) {
+        BLangExpression expr = (BLangExpression) varRefExpr.getExpression();
+        if (expr.getKind() == NodeKind.LITERAL) {
+            expr.accept(this);
+        } else if (expr.getKind() == NodeKind.RECORD_LITERAL_EXPR) {
+            expr.accept(this);
+        } else if (expr.getKind() == NodeKind.LIST_CONSTRUCTOR_EXPR) {
+            expr.accept(this);
+        }
+    }
+
+    @Override
+    public void visit(BLangListConstructorExpr listConstructorExpr) {
+        List<BLangExpression> exprList = listConstructorExpr.getExpressions();
+
+        for (BLangExpression expr : exprList) {
+            if (expr.getKind() == NodeKind.RECORD_LITERAL_EXPR) {
+                expr.accept(this);
+            } else if (expr.getKind() == NodeKind.SIMPLE_VARIABLE_REF) {
+                //Note: Here we do not need to travers further because we know that this is an array
+                //and the type is known in advance in this method using listConstructorExpr.type
+            }
+        }
+    }
+
+    @Override
+    public void visit(BLangSimpleVariable varNode) {
+        structureMap.put(varNode.symbol.name.toString(), varNode.symbol.type.toString());
+    }
+
+    @Override
+    public void visit(BLangRecordLiteral.BLangRecordKeyValueField recordKeyValue) {
+        if (recordKeyValue.valueExpr.getKind() == NodeKind.LITERAL) {
+            ((BLangLiteral)recordKeyValue.valueExpr).accept(this);
+        } else if (recordKeyValue.valueExpr.getKind() == NodeKind.RECORD_LITERAL_EXPR) {
+            recordKeyValue.valueExpr.accept(this);
+        } else if (recordKeyValue.valueExpr.getKind() == NodeKind.LIST_CONSTRUCTOR_EXPR) {
+            recordKeyValue.valueExpr.accept(this);
+        } else if (recordKeyValue.valueExpr.getKind() == NodeKind.SIMPLE_VARIABLE_REF) {
+            recordKeyValue.valueExpr.accept(this);
+        } else if (recordKeyValue.valueExpr.getKind() == NodeKind.NUMERIC_LITERAL) {
+            recordKeyValue.valueExpr.accept(this);
+        }
+    }
+
+    @Override
+    public void visit(BLangRecordLiteral recordLiteral) {
+        for (Iterator<RecordLiteralNode.RecordField> iterator = recordLiteral.fields.iterator(); iterator.hasNext(); ) {
```

Comment:
Can't we replace this with a simple for-each loop. eg. `for(RecordLiteralNode.RecordField field : recordLiteral.fields){...}`

Reply:
Yes, we can use this approach. I update the code to follow this approach.